### ISSUE TITLE
checkSchema() typescript error

### ISSUE THREAD

**Issue by Otto-AA** [express-validator/express-validator#626]
When I try to use `checkSchema(schema)` I get a typescript error, that the schema type and the expected one are incompatible.

It seems to me, that `checkSchema` doesn't use the [ExpressValidator.ValidationSchema](https://github.com/express-validator/express-validator/blob/298177a7c85b90110fa480422ee3dde472fb3c14/index.d.ts#L35) type, but a [different one](https://github.com/express-validator/express-validator/blob/298177a7c85b90110fa480422ee3dde472fb3c14/check/schema.d.ts#L7) which is incompatible.

Do you know if I just made a mistake, or if there's a fix for this?

#### Setup environment
Run following commands to install the modules
```bash
npm init -y
npm install express express-validator typescript --save
```

Save this to a typescript file (e.g. schemaValidator.ts)
```typescript
import expressValidator from "express-validator";
import { checkSchema } from "express-validator/check";

const validationSchema: ExpressValidator.ValidationSchema = {};
checkSchema(validationSchema);
```

#### Run typescript
When I now run `tsc .\schemaValidator.ts` I get following error:
```
schemaValidator.ts:5:13 - error TS2345: Argument of type 'Record<string, ValidationSchemaParamOptions | { [custom: string]: ValidatorSchemaOptions; }>' is not assignable to parameter of type 'Record<string, ValidationParamSchema>'.
  Index signatures are incompatible.
    Type 'ValidationSchemaParamOptions | { [custom: string]: ValidatorSchemaOptions; }' is not assignable to type 'ValidationParamSchema'.
      Type 'ValidationSchemaParamOptions' is not assignable to type 'ValidationParamSchema'.
        Property 'in' is optional in type 'ValidationSchemaParamOptions' but required in type 'ValidationParamSchema'.

5 checkSchema(validationSchema);
```

#### Versions:
- Express-validator: 5.3.0
- Typescript: 3.0.1
- Express: 4.16.3
- Node: 8.11.1
- Npm: 5.6.0

**Comment by Otto-AA** [comment]
Thanks, I've actually tried to use the type from /check, but as the example from the homepage didn't work with it, I thought that this would also be the wrong way. But using this resolved my issues (except for #618 ).

So my suggestion would be, to at least add a note for typescript in the schema validation tutorial (https://express-validator.github.io/docs/schema-validation.html) or even make a new site where a typescript compatible example is shown. This would help everyone who follows the tutorial.

You could also prefix the older type like `_ValidationSchema`. This way most developers would know to stay away from it. (I don't know if that is possible or exactly what you didn't want to do as it could break legacy code)

**Comment by gustavohenke** [comment]
express-validator is written in TypeScript now, so this should not be a problem anymore.
Please let me know if you still face it.
